bundles/org.openhab.binding.nobohub/src/main/java/org/openhab/binding/nobohub/internal/ComponentHandler.java

```diff
@@ -0,0 +1,160 @@
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Shows information about a Component in the Nobø Hub.
+ *
+ * @author Jørgen Austvik - Initial contribution
+ */
+@NonNullByDefault
+public class ComponentHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ComponentHandler.class);
+
+    private final NoboHubTranslationProvider messages;
+
+    protected @Nullable SerialNumber serialNumber;
+
+    public ComponentHandler(Thing thing, NoboHubTranslationProvider messages) {
+        super(thing);
+        this.messages = messages;
+    }
+
+    public void onUpdate(Component component) {
+        updateStatus(ThingStatus.ONLINE);
+
+        double temp = component.getTemperature();
+        if (!Double.isNaN(temp)) {
+            try {
+                QuantityType<Temperature> currentTemperature = new QuantityType<>(temp, SIUnits.CELSIUS);
+                updateState(CHANNEL_COMPONENT_CURRENT_TEMPERATURE, currentTemperature);
+            } catch (NumberFormatException nfe) {
+                logger.debug("Couldn't set decimal value to temperature: {}", temp);
+            }
+        }
+
+        updateProperty(Thing.PROPERTY_SERIAL_NUMBER, component.getSerialNumber().toString());
+        updateProperty(NAME, component.getName());
+        updateProperty(MODEL, component.getSerialNumber().getComponentType());
+
+        String zoneName = getZoneName(component.getZoneId());
+        if (zoneName != null) {
+            updateProperty(ZONE, zoneName);
+        }
+
+        String tempForZoneName = getZoneName(component.getTemperatureSensorForZoneId());
+        if (tempForZoneName != null) {
+            updateProperty(TEMPERATURE_SENSOR_FOR_ZONE, tempForZoneName);
+        }
+    }
+
+    private @Nullable String getZoneName(int zoneId) {
+        Bridge noboHub = getBridge();
+        if (null != noboHub) {
+            NoboHubBridgeHandler hubHandler = (NoboHubBridgeHandler) noboHub.getHandler();
+            if (hubHandler != null) {
+                Zone zone = hubHandler.getZone(zoneId);
+                if (null != zone) {
+                    return zone.getName();
+                }
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public void initialize() {
+        String serialNumberString = getConfigAs(ComponentConfiguration.class).serialNumber;
+        if (serialNumberString != null && !serialNumberString.isEmpty()) {
+            SerialNumber sn = new SerialNumber(serialNumberString);
+            if (!sn.isWellFormed()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        messages.get("component.illegal.serial", serialNumber));
```

Comment:
For `updateStatus` you can directly use "@text/component.illegal.serial" (assuming that's the i18n key) without any need for custom code.

Reply:
Example:
https://github.com/openhab/openhab-addons/blob/9f8c1772d2e4a93574c103a0bdcd491d6e78b2eb/bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/handler/DanfossAirUnitHandler.java#L110-111

Reply:
Good tip. This fixes messages with one argument, but I could not get it to work with two arguments, either way an improvement.